### PR TITLE
[AIRFLOW-XXXX] Add cross-reference between connection and fernet

### PR DESCRIPTION
**Pull request by mik-laj**
The `Manage connection` page contains very detailed information. However, it doesn't contain information about fernet.
 
---
Issue link: `Document only change, no JIRA issue`

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: docs/howto/connection/index.rst [hunk 1]
```diff
@@ -221,6 +221,13 @@ To fix this, you can encode with :py:meth:`~urllib.parse.quote_plus`:
     >>> print(c.password)
     my-pa/ssword
 
+Securing Connections
+--------------------
+
+Airflow uses `Fernet <https://github.com/fernet/spec/>`__ to encrypt passwords in the connection
+configuration. It guarantees that a password encrypted using it cannot be manipulated or read without the key.
+
+For information on configuring Fernet, look at :ref:`Generating a Connection URI <security/fernet>`.
```
Comment (kaxil) [inline docs/howto/connection/index.rst]: We had https://airflow.apache.org/docs/stable/howto/secure-connections.html#securing-connections . Did these get lost in the refactor?

Comment (mik-laj) [reply docs/howto/connection/index.rst]: It has been moved to security page: 
https://airflow.readthedocs.io/en/latest/security.html#securing-connections
Now I want to add a link that points to the ["Security" page](https://airflow.readthedocs.io/en/latest/security.html#securing-connections)  to the ["Managing Connections" page](https://airflow.readthedocs.io/en/latest/howto/connection/index.html).  Many people may not notice the Fernet configuration if it is on the Security page only.

Comment (mik-laj) [reply docs/howto/connection/index.rst]: I would like to divide the security page into several smaller subpages, but this is a different task.

Comment (mik-laj) [reply docs/howto/connection/index.rst]: The similar reference we have on "[Using the CLI](https://airflow.readthedocs.io/en/latest/usage-cli.html#creating-a-connection)" page. We have many guides on this page, but guide about connections on "[Managing Connections](https://airflow.readthedocs.io/en/latest/howto/connection/index.html#connection-cli)" page.

File: docs/howto/connection/index.rst [hunk 2]
```diff
@@ -221,6 +221,13 @@ To fix this, you can encode with :py:meth:`~urllib.parse.quote_plus`:
     >>> print(c.password)
     my-pa/ssword
 
+Securing Connections
+--------------------
+
+Airflow uses `Fernet <https://github.com/fernet/spec/>`__ to encrypt passwords in the connection
+configuration. It guarantees that a password encrypted using it cannot be manipulated or read without the key.
```
Comment (kaxil) [inline docs/howto/connection/index.rst]:
```suggestion
configuration. It guarantees that without the encryption password, Connection Passwords cannot be manipulated or read without the key.
```

Comment (mik-laj) [reply docs/howto/connection/index.rst]: Done.

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**